src/main/java/com/samoilenko/onlinebookstore/controller/AuthentificationController.java

```diff
@@ -0,0 +1,25 @@
+package com.samoilenko.onlinebookstore.controller;
+
+import com.samoilenko.onlinebookstore.dto.UserRegistrationRequestDto;
+import com.samoilenko.onlinebookstore.dto.UserResponseDto;
+import com.samoilenko.onlinebookstore.exception.RegistrationException;
+import com.samoilenko.onlinebookstore.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/api/auth")
+public class AuthentificationController {
+    private final UserService userService;
+
+    @PostMapping("/registration")
+    public UserResponseDto register(@RequestBody UserRegistrationRequestDto request)
```

Comment:
did you test your validation? because it does not work without this:
```suggestion
    public UserResponseDto register(@RequestBody @Valid UserRegistrationRequestDto request)
```